packages/pci-sdk-iframe/src/apiClient/index.ts

```diff
@@ -43,10 +43,12 @@ interface IVerify2FACodeResponse {
 }
 
 export async function verify2FACode(secret: string, verificationId: string): Promise<IVerify2FACodeResponse> {
+	const santizedSecret = formatterService.sanitize2FACode(secret);
+
 	return fetch(`${BASE_URL}v1/verifications/${verificationId}/finish`, {
 		method: 'POST',
 		headers,
-		body: JSON.stringify({ secret }),
+		body: JSON.stringify({ santizedSecret }),
```

Comment:
Is this even working? ¬¬

you are sending a totally new object with this structure:

```ts
{
  "santizedSecret" : "whatever"
}
```

while the server expects 

```ts
{
  "secret" :  "whatever"
}
```

This is worrying.

I know that being a matter of integration with the API we do not have automatic tests. But you should have tested it manually or running cypress E2E. What happened? Didn't you run the tests or the tests didn't detect the error?

Reply:
Bah, you are right. On first commit I was sanitizing the code in the hook and then, just before logging off for the day I hastily moved the logic to the API. I reran the unit tests but not the cypress tests... maybe I should see why it wasn't caught by the unit tests.